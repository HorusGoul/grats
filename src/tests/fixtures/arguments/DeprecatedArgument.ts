/** @gqlType */
export default class Query {
  /** @gqlField */
  hello({
    greeting,
  }: {
    /** @deprecated Not used anymore */
    greeting?: string | null;
  }): string {
    return "Hello world!";
  }
}
