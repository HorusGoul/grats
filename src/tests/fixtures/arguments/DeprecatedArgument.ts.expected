-----------------
INPUT
----------------- 
/** @gqlType */
export default class SomeType {
  /** @gqlField */
  hello({
    greeting,
  }: {
    /** @deprecated Not used anymore */
    greeting?: string | null;
  }): string {
    return "Hello world!";
  }
}

-----------------
OUTPUT
-----------------
directive @asyncIterable on FIELD_DEFINITION

directive @methodName(name: String!) on FIELD_DEFINITION

directive @exported(jsModulePath: String!, tsModulePath: String!, functionName: String!, argCount: Int!) on FIELD_DEFINITION

type SomeType {
  hello(greeting: String @deprecated(reason: "Not used anymore")): String
}